### PR TITLE
[clang-tidy][abseil-string-find-startswith] Add string_view to default string-like classes

### PR DESCRIPTION
**Pull request by nicovank**
As per title. A small improvement to this check, `string_view` should work out of the box.

I am slowly looking into adding an option to this check to recommend C++20's API `starts_with`, and move it to `modernize`.
Possibly would like to use it as a codemod on our code.

### CODE REVIEW

File: clang-tools-extra/test/clang-tidy/checkers/abseil/string-find-startswith.cpp
Comment (nicovank) [inline clang-tools-extra/test/clang-tidy/checkers/abseil/string-find-startswith.cpp:38]: https://github.com/llvm/llvm-project/blob/57dd23bc0a2f7b4f7b68162923b3267c1f303de9/clang-tools-extra/test/clang-tidy/checkers/abseil/string-find-str-contains.cpp#L24-L37

**Comment by nicovank** [comment]
First time with LLVM GitHub PR workflow. Am I supposed to tag potential reviewers? @gribozavr @hokein @njames93

File: clang-tools-extra/test/clang-tidy/checkers/abseil/string-find-startswith.cpp
Comment (PiotrZSL) [inline clang-tools-extra/test/clang-tidy/checkers/abseil/string-find-startswith.cpp:25]: Actually string and string_view is defined already in header test/clang-tidy/checkers/Inputs/Headers/string.

Just add `-- -isystem %clang_tidy_headers` like tests for readability-string-compare does, and simply include string.

**Review by PiotrZSL — COMMENTED** [review]
As for this check:
- use string header
- update check documentation (option default and description)
- make sure that documentation is in sync with class documentation
- add release notes entry about improving check to support std::string_view

**Review by PiotrZSL — CHANGES_REQUESTED** [review]
documentation, string header, release notes

**Comment by nicovank** [comment]
> Copy this check.

Got it. Thanks for the pointer!

> Documentation, string header.

I think I covered everything, had to add a couple of skeletons to that `string` common header.

File: clang-tools-extra/clang-tidy/abseil/StringFindStartswithCheck.cpp
Comment (PiotrZSL) [inline clang-tools-extra/clang-tidy/abseil/StringFindStartswithCheck.cpp:23]: Remove this space
```suggestion
    "::std::basic_string;::std::basic_string_view";
```

File: clang-tools-extra/docs/clang-tidy/checks/abseil/string-find-startswith.rst
Comment (PiotrZSL) [inline clang-tools-extra/docs/clang-tidy/checks/abseil/string-find-startswith.rst:31]: Line 6 in this file should also be changed, something like:
"Checks whether a ``std::string::find()`` or ``std::string::rfind()``  (and corresponding ``std::string_view`` methods)"

Simply should be there some info also about string_view, no need to update examples, just description.

**Review by PiotrZSL — APPROVED** [review]
LGTM, once you correct latest comments & CI will pass, fell free to merge.

**Comment by nicovank** [comment]
Addressed feedback.

I do not have commit access, can you please merge after CI is green?
Assuming GitHub will properly populate information, if not you can use `Nicolas van Kempen <nvankempen@meta.com>`.

**Pull request closed without merging**
